From cdc59e97914702aa57eb4a41aa909c182f749444 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 17 Feb 2021 00:43:26 +0200
Subject: [PATCH] [SW-29740] gaudi: security: introduce lbw rrs enum

Since we are now using the RRs numbers in more than 1 place (e.g.
preboot_secure_lbw() and preboot_security_dbi_lock(), we should be using
the same RR number w/o the need to update both places.
Introduce the gaudi_sec_lbw_rrs enum to reference the RRs by name.
This also lets other RRs benefit from the enum.
Additionally, the RRs numbers are effectively changed to be configuous.

Change-Id: I62c1ceda052fe02b8131f63f244f182c21daab9f
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 45ed860..5d41ab2 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1166,6 +1166,21 @@
 	preboot_secure_pcie_rr(rr, min, max);
 }
 
+enum gaudi_sec_lbw_rrs {
+	LBW_RR_SRAM_CFG,
+	LBW_RR_CPU_TSTAMP,
+	LBW_RR_HBM_GICBGN,
+	LBW_RR_GICEND,
+	LBW_RR_PCIE_DBI_COREBGN,
+	LBW_RR_PCIE_COREEND_AUX_PHY,
+	LBW_RR_PSOC0,
+	LBW_RR_PSOC1_PCIE_PMA,
+	LBW_RR_NIC0,
+	LBW_RR_NIC1,
+	LBW_RR_NIC3,
+	LBW_RR_ETR,
+};
+
 #define SYNC_MNGR_DIST		0x20000
 #define SYNC_MNGR_SIZE		0x10000
 
@@ -1179,70 +1194,70 @@
 	 * meaning - not inclusive, so we need to provide it the min address
 	 * below the forbidden one and max address above the forbidden one.
 	 */
-	/* SRAM_CFG up to SIF: RR_0 */
+	/* SRAM_CFG up to SIF: RR(LBW_RR_SRAM_CFG) */
 	min = mmSRAM_Y0_X0_BANK_BASE - 1;
 	max = mmSIF_RTR_0_BASE;
-	preboot_secure_lbw_rr(0, min, max);
+	preboot_secure_lbw_rr(LBW_RR_SRAM_CFG, min, max);
 
-	/* CPU_TIMESTAMP: RR_1 */
+	/* CPU_TIMESTAMP: RR(LBW_RR_CPU_TSTAMP) */
 	min = mmCPU_TIMESTAMP_BASE - 1;
 	max = mmDMA_IF_W_S_BASE;
-	preboot_secure_lbw_rr(1, min, max);
+	preboot_secure_lbw_rr(LBW_RR_CPU_TSTAMP, min, max);
 
-	/* HBM_CFG up to GIC_L: RR_6 */
+	/* HBM_CFG up to GIC_L: RR(LBW_RR_HBM_GICBGN) */
 	min = mmHBM0_BASE - 1;
 	max = HL_CFG_BASE + mmGIC_DISTRIBUTOR__5_GICD_SETSPI_NSR;
-	preboot_secure_lbw_rr(6, min, max);
+	preboot_secure_lbw_rr(LBW_RR_HBM_GICBGN, min, max);
 
-	/* GIC_H up to PCIE_WRAP: RR_7 */
+	/* GIC_H up to PCIE_WRAP: RR(LBW_RR_GICEND) */
 	min = max + sizeof(u32) - 1;
 	max = mmPCIE_WRAP_BASE;
-	preboot_secure_lbw_rr(7, min, max);
+	preboot_secure_lbw_rr(LBW_RR_GICEND, min, max);
 
 	/*
 	 * Note: we skip here the PCIE_DBI, since iATU might not be configured
 	 * yet and thus host needs access to DBI/iATU space.
 	 * The DBI is closed at a later stage in DBI lock function.
 	 */
-	/* PCIE_CORE up to PCIE_CORE_L: RR_8 */
+	/* PCIE_CORE up to PCIE_CORE_L: RR(LBW_RR_PCIE_DBI_COREBGN) */
 	min = mmPCIE_CORE_BASE - 1;
 	max = HL_CFG_BASE + mmPCIE_CORE_MSI_REQ;
-	preboot_secure_lbw_rr(8, min, max);
+	preboot_secure_lbw_rr(LBW_RR_PCIE_DBI_COREBGN, min, max);
 
-	/* PCIE_CORE_H up to MMU_UP: RR_9 */
+	/* PCIE_CORE_H up to MMU_UP: RR(LBW_RR_PCIE_COREEND_AUX_PHY) */
 	min = max + sizeof(u32) - 1;
 	max = mmMMU_UP_BASE;
-	preboot_secure_lbw_rr(9, min, max);
+	preboot_secure_lbw_rr(LBW_RR_PCIE_COREEND_AUX_PHY, min, max);
 
-	/* PSOC_I2C_M0 up to PSOC_TIMESTAMP: RR_10 */
+	/* PSOC_I2C_M0 up to PSOC_TIMESTAMP: RR(LBW_RR_PSOC0) */
 	min = mmPSOC_I2C_M0_BASE - 1;
 	max = mmPSOC_TIMESTAMP_BASE;
-	preboot_secure_lbw_rr(10, min, max);
+	preboot_secure_lbw_rr(LBW_RR_PSOC0, min, max);
 
-	/* PSOC_GPIO0 up to NIC0_MAC_CH0: RR_11 */
+	/* PSOC_GPIO0 up to NIC0_MAC_CH0: RR(LBW_RR_PSOC1_PCIE_PMA) */
 	min = mmPSOC_GPIO0_BASE - 1;
 	max = mmNIC0_MAC_CH0_BASE;
-	preboot_secure_lbw_rr(11, min, max);
+	preboot_secure_lbw_rr(LBW_RR_PSOC1_PCIE_PMA, min, max);
 
-	/* NIC0_TS up to NIC1_MAC_CH0: RR_12 */
+	/* NIC0_TS up to NIC1_MAC_CH0: RR(LBW_RR_NIC0) */
 	min = mmNIC0_TS_BASE - 1;
 	max = mmNIC1_MAC_CH0_BASE;
-	preboot_secure_lbw_rr(12, min, max);
+	preboot_secure_lbw_rr(LBW_RR_NIC0, min, max);
 
-	/* NIC1_TS up to NIC2_MAC_CH0: RR_13 */
+	/* NIC1_TS up to NIC2_MAC_CH0: RR(LBW_RR_NIC1) */
 	min = mmNIC1_TS_BASE - 1;
 	max = mmNIC2_MAC_CH0_BASE;
-	preboot_secure_lbw_rr(13, min, max);
+	preboot_secure_lbw_rr(LBW_RR_NIC1, min, max);
 
-	/* NIC3_TS up to NIC4_MAC_CH0: RR_14 */
+	/* NIC3_TS up to NIC4_MAC_CH0: RR(LBW_RR_NIC3) */
 	min = mmNIC3_TS_BASE - 1;
 	max = mmNIC4_MAC_CH0_BASE;
-	preboot_secure_lbw_rr(14, min, max);
+	preboot_secure_lbw_rr(LBW_RR_NIC3, min, max);
 
-	/* mmPSOC_ETR_AXICTL (single register): RR_15 */
+	/* mmPSOC_ETR_AXICTL (single register): RR(LBW_RR_ETR) */
 	min = HL_CFG_BASE + mmPSOC_ETR_AXICTL - 1;
 	max = HL_CFG_BASE + mmPSOC_ETR_AXICTL + sizeof(u32);
-	preboot_secure_lbw_rr(15, min, max);
+	preboot_secure_lbw_rr(LBW_RR_ETR, min, max);
 
 	/* SYNC_MNGR: PCIe only! PCIe RR_28 up to PCIe RR_31 */
 	for (i = 0; i < 4; i++) {
@@ -1621,8 +1636,8 @@
 	if (!preboot_security_is_enabled())
 		return;
 
-	/* PCIE_DBI up to PCIE_CORE_L: RR_8 */
+	/* PCIE_DBI up to PCIE_CORE_L: RR(LBW_RR_PCIE_DBI_COREBGN) */
 	min = mmPCIE_DBI_BASE - 1;
 	max = HL_CFG_BASE + mmPCIE_CORE_MSI_REQ;
-	preboot_secure_lbw_rr(8, min, max);
+	preboot_secure_lbw_rr(LBW_RR_PCIE_DBI_COREBGN, min, max);
 }
